src/model/application/interaction/message_component.rs

```diff
@@ -219,15 +235,32 @@ impl MessageComponentInteraction {
     {
         let mut interaction_response = CreateInteractionResponseFollowup::default();
         f(&mut interaction_response);
+        let message_id = message_id.into().into();
+        self._edit_followup_message(http.as_ref(), message_id, interaction_response).await
+    }
 
+    async fn _edit_followup_message<'a>(
+        &self,
+        http: &Http,
+        message_id: u64,
+        interaction_response: CreateInteractionResponseFollowup<'a>,
+    ) -> Result<Message> {
```

Comment:
Since `ApplicationCommandInteraction::_edit_followup_message` doesn't exist, I think that should be duplicated here, regardless of any extra monomorphization costs that might incur by calling everything inside the main `edit_followup_message` function. Besides, having `message_id: u64` isn't ideal (on `next` we've shifted to passing IDs around everywhere instead).